### PR TITLE
Move `QueryHistoryInfo` type to separate file

### PR DESCRIPTION
**Pull request by shati-patel**
I'm working on changing the methods that deal with `QueryHistoryInfo` (specifically: making sure that they handle the new `VariantAnalysisHistoryItem` type). See internal issue for more info.

Since the `query-results.ts` and `query-history.ts` files are getting pretty big and cluttered, I'd like to move the `QueryHistoryInfo` type and any associated helper functions to a new separate file. 

## Checklist

- [x] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [x] Issues have been created for any UI or other user-facing changes made by this pull request.
- [x] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/src/query-history.ts [hunk 1]
```diff
@@ -30,7 +30,8 @@ import { DisposableObject } from './pure/disposable-object';
 import { commandRunner } from './commandRunner';
 import { ONE_HOUR_IN_MS, TWO_HOURS_IN_MS } from './pure/time';
 import { assertNever, getErrorMessage, getErrorStack } from './pure/helpers-pure';
-import { CompletedLocalQueryInfo, LocalQueryInfo as LocalQueryInfo, QueryHistoryInfo } from './query-results';
+import { CompletedLocalQueryInfo, LocalQueryInfo as LocalQueryInfo } from './query-results';
```
Comment (charisk) [inline extensions/ql-vscode/src/query-history.ts:33]:
```suggestion
import { CompletedLocalQueryInfo, LocalQueryInfo } from './query-results';
```

File: extensions/ql-vscode/src/query-history-info.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { RemoteQueryHistoryItem } from './remote-queries/remote-query-history-item';
+import { VariantAnalysisHistoryItem } from './remote-queries/variant-analysis-history-item';
+import { LocalQueryInfo } from './query-results';
+import { assertNever } from './pure/helpers-pure';
+
+export type QueryHistoryInfo = LocalQueryInfo | RemoteQueryHistoryItem | VariantAnalysisHistoryItem;
+
+export function getRawName(item: QueryHistoryInfo): string {
```
Comment (charisk) [inline extensions/ql-vscode/src/query-history-info.ts:8]: Why is it `raw`? 🤔 I'd call this `gerQueryName` or `getRawQueryName`

**Review by charisk — APPROVED** [review]
Thanks for tidying that up! Just a couple of small comments.

File: extensions/ql-vscode/src/query-history-info.ts (hunk 2 above)
Comment (charisk) [inline extensions/ql-vscode/src/query-history-info.ts:8]: We should probably add some unit tests around these functions too.

Comment (shati-patel) [reply extensions/ql-vscode/src/query-history-info.ts]: Yes! I'll pull those changes (from our pairing session) to a follow-up PR if that's okay 🧹

**Review by charisk — APPROVED** [review]

**Pull request merged**